core/templates/dev/head/components/ruleEditor.js

```diff
@@ -153,15 +153,25 @@ oppia.directive('ruleEditor', ['$log', function($log) {
             // This rule is for a multiple-choice or image-click interaction.
             // TODO(sll): Remove the need for this special case.
             if (_answerChoices.length > 0) {
-              $scope.ruleDescriptionChoices = _answerChoices.map(function(choice, ind) {
-                return {
-                  val: choice.label,
-                  id: choice.val
-                };
-              });
-              result.push({'type': 'select', 'varName': finalInputArray[i+1]});
-              if (!$scope.rule.inputs[finalInputArray[i + 1]]) {
-                $scope.rule.inputs[finalInputArray[i + 1]] = $scope.ruleDescriptionChoices[0].id;
+              if (finalInputArray[2] === 'SetOfHtmlString') {
+                $scope.ruleDescriptionChoices = _answerChoices.map(function(choice, ind) {
+                  return {
+                    val: false,
```

Comment:
Hmm. Here and in the next 'else' clause, you are conveying the concept of 'choices' -- but the return values look quite different. I think they should be more similar, and that the conversion to checkbox form should happen closer to the checkbox part of the code.